### PR TITLE
Articles search by phrase

### PR DESCRIPTION
**Pull request by cedrospl**
Save articles' links to the list
Open new tab with links from the list
Save the list to the file

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -17,31 +17,31 @@
...
-            <version>3.141.59</version>
+            <version>4.7.1</version>
         </dependency>
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
-            <version>4.12</version>
-            <scope>test</scope>
+            <version>4.13.2</version>
+            <scope>compile</scope>
```
Comment (ajjankowski) [inline pom.xml]: It's doubled

File: pom.xml [hunk 2]
```diff
@@ -58,13 +58,13 @@
         <dependency>
             <groupId>org.seleniumhq.selenium</groupId>
             <artifactId>selenium-api</artifactId>
-            <version>3.141.59</version>
+            <version>4.7.1</version>
             <scope>compile</scope>
```
Comment (ajjankowski) [inline pom.xml]: It's doubled

File: src/main/pages/HomePageOnet.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        return homeWeatherText.getText();
+    }
+
+    public String acceptCookies2() {
+        homeCookieFirstStepAcceptButton.click();
+        homeCookieSecondStepAcceptButton.click();
+        return getLogoText();
+    }
+
+    public void getArticleLinksOpenThemInDifferentTabsAndSaveLinkstoFile() throws IOException {
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: Too long, it's not german-coding xD

File: src/main/pages/HomePageOnet.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        System.out.println("Saving article links to the list");
+        List<WebElement> phraseSearch2 = driver.findElements(By.partialLinkText("ska"));
+        WebElement q;
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: ? / var has to be distinguishable

File: src/main/pages/HomePageOnet.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        String k;
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: ? / var has to be distinguishable

File: src/test/OnetTest.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+
+public class OnetTest extends BaseTest {
+    @Test
+    public void testOpenBrowserWithOnet() throws IOException {
+
```
Comment (ajjankowski) [inline src/test/OnetTest.java]: Unnecessary line

File: src/main/pages/HomePageOnet.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+            q = webElement;
+            k = q.getAttribute("href");
+            articleLinks.add(k);
+        }
+        System.out.println("Opening a new tab with the article link from the list");
+        for (String articleLink : articleLinks) {
+            driver.findElement(By.cssSelector("body")).sendKeys(Keys.CONTROL + "t");
+            driver.get(String.valueOf(articleLink));
+
+        }
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: Code is opening all sites in one tab, one after another. It's hard to check the results. It should open each in a new tab.

File: src/main/pages/HomePageOnet.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        for (WebElement webElement : phraseSearch2) {
+            q = webElement;
+            k = q.getAttribute("href");
+            articleLinks.add(k);
+        }
+        System.out.println("Opening a new tab with the article link from the list");
+        for (String articleLink : articleLinks) {
+            driver.findElement(By.cssSelector("body")).sendKeys(Keys.CONTROL + "t");
+            driver.get(String.valueOf(articleLink));
+
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: Redundant line

File: src/main/pages/HomePageOnet.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        System.out.println("Saving Article Links ");
+        Path dest = Paths.get("SavedArticleLinks.java"); // można też podać patha do innego folderu w projekcie
+        try {
+            Path p = Files.write(dest, articleLinks, StandardOpenOption.WRITE, StandardOpenOption.CREATE);
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: "p" - as above

File: src/main/pages/HomePageOnet.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+            articleLinks.add(k);
+        }
+        System.out.println("Opening a new tab with the article link from the list");
+        for (String articleLink : articleLinks) {
+            driver.findElement(By.cssSelector("body")).sendKeys(Keys.CONTROL + "t");
+            driver.get(String.valueOf(articleLink));
+
+        }
+        System.out.println("Saving Article Links ");
+        Path dest = Paths.get("SavedArticleLinks.java"); // można też podać patha do innego folderu w projekcie
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: Save it to *.csv or *.txt file, it will be easier later on. This class will be a useful util.

**Review by ajjankowski — CHANGES_REQUESTED** [review]
1) After dealing with comments, we can discuss approach for for-each loops or using streams.
2) Change tree structure, right now I have to struggle to run it. After this, it would be good to add in ReadMe which file should be Test source root / main. It would be best to keep all in one Test source root "test" not using java root for such a small project.

File: src/main/pages/HomePageOnet.java (hunk 3 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: Fixed in the newest commit.

File: src/main/pages/HomePageOnet.java (hunk 4 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: Fixed in the newest commit.

File: src/main/pages/HomePageOnet.java (hunk 5 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: Fixed in the newest commit.

File: src/test/OnetTest.java (hunk 6 above)
Comment (cedrospl) [reply src/test/OnetTest.java]: Fixed in the newest commit.

File: src/main/pages/HomePageOnet.java (hunk 7 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: Fixed in the newest commit.

File: src/main/pages/HomePageOnet.java (hunk 8 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: Fixed in the newest commit.

File: src/main/pages/HomePageOnet.java (hunk 9 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: Fixed in the newest commit.

File: src/main/pages/HomePageOnet.java (hunk 10 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: Fixed in the newest commit.

File: src/main/pages/HomePageOnet.java [hunk 11]
```diff
@@ -55,30 +52,35 @@ public String acceptCookies2() {
...
         System.out.println("Saving article links to the list");
-        List<WebElement> phraseSearch2 = driver.findElements(By.partialLinkText("ska"));
-        WebElement q;
-        String k;
+        /**
+         Change "ska" to whatever text you want to search it on www.onet.pl site
+         */
+        List<WebElement> searchByPhrase = driver.findElements(By.partialLinkText("ska"));
+        WebElement phraseSearch;
+        String singleArticle;
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: Redundant variables, delete

File: src/main/pages/HomePageOnet.java [hunk 12]
```diff
@@ -55,30 +52,35 @@ public String acceptCookies2() {
...
         List<String> articleLinks = new ArrayList<>();
-        for (WebElement webElement : phraseSearch2) {
-            q = webElement;
-            k = q.getAttribute("href");
-            articleLinks.add(k);
+        for (WebElement webElement : searchByPhrase) {
+            phraseSearch = webElement;
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: Redundant, delete

File: src/main/pages/HomePageOnet.java [hunk 13]
```diff
@@ -55,30 +52,35 @@ public String acceptCookies2() {
...
+            singleArticle = phraseSearch.getAttribute("href");
```
Comment (ajjankowski) [inline src/main/pages/HomePageOnet.java]: Change to: "String singleArticle = webElement.getAttribute("href");"
WebElement is declared in for loop, String can be here. No need for reassigning variables in code.

**Review by ajjankowski — CHANGES_REQUESTED** [review]
1) Add in 1st line of "BaseTest" class: "package test;"
2) Before "driver = new ChromeDriver();" in line 13 of "BaseTest" add: "WebDriverManager.chromedriver().setup();" After using this new library, downloading and updating ChromeDriver will be needed no more :)
3) In "BaseTest" delete tearDown, we open new tabs to read them :P

File: src/main/pages/HomePageOnet.java (hunk 12 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: You mean here delete lines 64 and 65? I cannot delete this loop, it searches for articles on the page :-D
If you mean delete lines 66 and 67 it's already deleted in this commit.

File: src/main/pages/HomePageOnet.java (hunk 11 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: That's true, variables are redundant and already deleted (change will appear in the next commit).

File: src/main/pages/HomePageOnet.java (hunk 13 above)
Comment (cedrospl) [reply src/main/pages/HomePageOnet.java]: Nice, change is made (change will appear in the next commit).

File: src/main/pages/HomePageOnet.java (hunk 12 above)
Comment (ajjankowski) [reply src/main/pages/HomePageOnet.java]: Just 65 - my bad

**Review by ajjankowski — APPROVED** [review]

**Pull request merged**